plutus-core/plutus-core/src/PlutusCore/MkPlc.hs

```diff
@@ -210,23 +213,39 @@ mkImmediateTyAbs
 mkImmediateTyAbs ann1 (Def (TyVarDecl ann2 name k) bind) body =
     tyInst ann1 (tyAbs ann2 name k body) bind
 
--- | Make an iterated application.
+-- | Make an iterated application. Each `apply` node uses the annotation associated with
+-- the corresponding argument.
 mkIterApp
     :: TermLike term tyname name uni fun
-    => ann
-    -> term ann -- ^ @f@
-    -> [term ann] -- ^@[ x0 ... xn ]@
-    -> term ann -- ^ @[f x0 ... xn ]@
-mkIterApp ann = foldl' (apply ann)
+    => term ann
+    -> [(ann, term ann)]
+    -> term ann
+mkIterApp = foldl' (uncurry . flip apply)
```

Comment:
Nit: at that point I'd add some names, `uncurry`, `.` and `flip` together is a bit too much even for a point-free abuser like myself.

Reply:
Understandable. I find it easier to read though :smile:

Reply:
> I find it easier to read though

I'm kind of amazed if this is true :D

Reply:
I mean, `uncurry` and `flip` are just restructuring the arguments and can be ignored, so it's really just `fold' apply`.